File: Data Structures & Algorithms/is-subsequence/submission-1.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        if not s:
            return True

        if not t:
            return False

        i = 0

        
        for char in t:
            if s[i] == char:
                i = i + 1

            if i == len(s):
                return True

        return i == len(s)

File: Data Structures & Algorithms/max-consecutive-ones/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    findMaxConsecutiveOnes(nums) {
        let max = 0
        let current = 0

        for (const num of nums) {
            current = num === 1 ? current + 1 : 0
            max = Math.max(current, max)
        }

        return max
    }
}
